kernel/sys/early_panic.cpp

```diff
@@ -12,9 +12,12 @@
 #include <arch/arch.hpp>
 #include <dev/tty/tty.hpp>
 
+// Provide a function prototype to make the compiler warnings happy. Don't
+// want to make it public though, so we won't put it in the panic header.
+extern "C" void early_panic(const char *str);
+
 extern "C" void
 __attribute__ ((section(".early_text")))
-__attribute__ ((optnone))
```

Comment:
Why was this removed? Won't that break this functionality?

Reply:
GCC complained about it. Apparently it's a clang only thing.

Reply:
Okay so we'll need to add the appropriate attribute for GCC. I believe it is something like `__attribute__((-O0))`.

Reply:
`__attribute__((-O0))` doesn't work. I'll see if I can't find what it is.

Reply:
Is `__attribute__((optimize("O0")))` what you're looking for?

Reply:
Probs

Reply:
Yeah, it is. Found a stackoverflow post about it.